docs/testing/testing.md

```diff
@@ -0,0 +1,293 @@
+# Intro to testing
+
+Before starting with a quick introduction to testing, we want to make sure you
+install `pytest`, the Python testing tool that we will use for writing our
+tests. 
+
+If you haven't installed `pytest`, you can do it using conda, open a terminal
+and type:
+
+```text
+$ conda install pytest
+```
+
+Alright! Now we can start by writing some code and some tests for it. If you
+want to read more about the documentation of `pytest` here is a [link](http://doc.pytest.org/en/latest/contents.html) where you can find all its 
+documentation.
+
+## Testing a dot_product function
+
+First of all let's open a terminal and create a new file called
+`my_dot_product.py`. If you are using `nano` type in a terminal:
```

Comment:
below we're running `dot_product.py` but up here it's `my_dot_product.py` -- I'm fine with either but it should be consistent.